src/field.rs

```diff
@@ -228,46 +253,114 @@ mod test {
         }
     }
 
+    // For working around the orphan trait impl rule
+    #[derive(Clone, Debug, PartialEq, Eq)]
+    pub struct VecFWrap<F: LurkField>(pub Vec<F>);
+
+    impl<F: LurkField> Arbitrary for VecFWrap<F> {
+        fn arbitrary(g: &mut Gen) -> Self {
+            let vec_f: Vec<FWrap<F>> = Arbitrary::arbitrary(g);
+            VecFWrap(vec_f.into_iter().map(|f| f.0).collect())
+        }
+    }
+
     #[quickcheck]
-    fn test_bytes_consistency(f1: FWrap<Fr>) -> bool {
+    fn prop_blstrs_repr_bytes_consistency(f1: FWrap<Fr>) -> bool {
         let bytes = f1.0.to_repr().as_ref().to_owned();
         let f2 = <Fr as LurkField>::from_bytes(&bytes);
         Some(f1.0) == f2
     }
 
     #[quickcheck]
-    fn test_tag_consistency(x: Tag) -> bool {
-        let f1 = Fr::from(x as u64);
-        let tag = <Fr as LurkField>::to_tag(f1).unwrap();
-        let f2 = Fr::from(tag as u64);
-        f1 == f2 && x as u32 == tag
+    fn prop_pallas_repr_bytes_consistency(f1: FWrap<pasta_curves::Fp>) -> bool {
+        let bytes = f1.0.to_repr().as_ref().to_owned();
+        let f2 = <pasta_curves::Fp as LurkField>::from_bytes(&bytes);
+        Some(f1.0) == f2
+    }
+
+    #[quickcheck]
+    fn prop_vesta_repr_bytes_consistency(f1: FWrap<pasta_curves::Fq>) -> bool {
+        let bytes = f1.0.to_repr().as_ref().to_owned();
+        let f2 = <pasta_curves::Fq as LurkField>::from_bytes(&bytes);
+        Some(f1.0) == f2
+    }
```

Comment:
If quickcheck (or later proptest) allows it, might be nicer with all of these tests (below also) to define a helper with type parameter, then call once for each field type. Especially if/when we add fields, that will keep things simpler.